pandas/core/groupby/groupby.py

```diff
@@ -3290,6 +3304,102 @@ def max(
                 npfunc=np.max,
             )
 
+    def _idxmin_idxmax(
+        self,
+        how: Literal["idxmax", "idxmin"],
+        numeric_only: bool = False,
+        skipna: bool = True,
+        ignore_unobserved: bool = False,
+    ) -> NDFrameT:
+        """Compute idxmax/idxmin.
+
+        Parameters
+        ----------
+        how: {"idxmin", "idxmax"}
+            Whether to compute idxmin or idxmax.
+        numeric_only : bool, default False
+            Include only float, int, boolean columns.
+        skipna : bool, default True
+            Exclude NA/null values. If an entire row/column is NA, the result
+            will be NA.
+        ignore_unobserved : bool, default False
+            When True and an unobserved group is encountered, do not raise. This used
+            for transform where unobserved groups do not play an impact on the result.
+
+        Returns
+        -------
+        Series or DataFrame
+            idxmax or idxmin for the groupby operation.
+        """
+        # seen is passed to the Cython code and mutated
+        seen = np.zeros(self.ngroups, dtype="bool")
```

Comment:
If you set this to all ones then set to zero in the loop you could use short-circuiting `.any()` calls later instead of having to do `not seen.all()`; depending on how often you think something would be "unseen" that could only help performance

Reply:
It would be unseen when `observed=False`, there is a single grouping (e.g. `.groupby('a')`), and there are unobserved categories. While not the common case, I can't see a reason not to do this.

Also, I've changed `seen -> unobserved`.